src/test/java/com/commercetools/pingpong/model/GameTest.java

```diff
@@ -0,0 +1,312 @@
+
+        for (dummyPlayerOne.getSetScore(); dummyPlayerOne.getSetScore() < setScore;) {
+            dummyPlayerOne.addPoint();
+        }
+
+        for (dummyPlayerOne.getMatchScore(); dummyPlayerOne.getMatchScore() < matchScore;) {
+            dummyPlayerOne.addMatchScore();
+        }
+
+        if (serve) {
+            dummyPlayerOne.setServe();
+        } else {
+            dummyPlayerOne.unsetServe();
+        }
+
+        if (beginningServe) {
+            dummyPlayerOne.setBeginningServe();
+        } else {
+            dummyPlayerOne.unsetBeginningServe();
+        }
+
+        return dummyPlayerOne;
+    }
+
+
+    public Player getDummyTwo(int setScore, int matchScore, boolean serve, boolean beginningServe) {
+        Player dummyPlayerTwo = new Player();
+
+        for (dummyPlayerTwo.getSetScore(); dummyPlayerTwo.getSetScore() < setScore;) {
+            dummyPlayerTwo.addPoint();
+        }
+
+        for (dummyPlayerTwo.getMatchScore(); dummyPlayerTwo.getMatchScore() < matchScore;) {
+            dummyPlayerTwo.addMatchScore();
+        }
+
+        if (serve) {
+            dummyPlayerTwo.setServe();
+        } else {
+            dummyPlayerTwo.unsetServe();
+        }
+
+        if (beginningServe) {
+            dummyPlayerTwo.setBeginningServe();
+        } else {
+            dummyPlayerTwo.unsetBeginningServe();
+        }
+
+        return dummyPlayerTwo;
+    }
+
+
+
+
+    @Test
+    public void shouldReturnNewGame() {
+        Player testPlayerOne = new Player("One");
+        Player testPlayerTwo = new Player("Two");
+
+        Game testGame = new Game(testPlayerOne, testPlayerTwo);
+
+        assertThat("It should return the name of the first player", testGame.getPlayerOne().getName(), is("One"));
+        assertThat("It should return the name of the second player", testGame.getPlayerTwo().getName(), is("Two"));
+
+    }
+
+    @Test
+    public void shouldAddSetScoreOfPlayer() {
+        Message testMessage = new Message("1", 1);
+        Player testplayer = new Player();
+        Game testMatch = new Game(testplayer, testplayer);
+
+        testMatch.updateScoreOfPlayer(testplayer, testMessage.getActionType());
+
+        assertThat("", testplayer.getSetScore(), is(1));
+
+    }
+
+
+    /** WARNING: If any of the following five tests fail, the above dummyPlayers methods do not work properly **/
+    @Test
+    public void shouldSubSetScoreOfPlayer() {
+        Message testMessageOne = new Message("1", 1);
+        Message testMesssageTwo = new Message("1", 2);
+        Player testplayer = new Player();
+        Game testMatch = new Game(testplayer, testplayer);
+
+        testMatch.updateScoreOfPlayer(testplayer, testMessageOne.getActionType());
+        testMatch.updateScoreOfPlayer(testplayer, testMessageOne.getActionType());
+        testMatch.updateScoreOfPlayer(testplayer, testMesssageTwo.getActionType());
+
+        assertThat("", testplayer.getSetScore(), is(1));
```

Comment:
It's good practice to provide an assertion message. Helps with looking and understanding the test assertions.

Reply:
Yes, I wrote the last ones that were still missing 👍 
2654df6c943c9878391e6a9a5fcc9488a0a38cde